doc/MATH.rst

```diff
@@ -0,0 +1,180 @@
+coordinate system that wouldn't otherwise be possible using vectors.
+
+Simple Movement
+^^^^^^^^^^^^^^^
+
+Lets take a few concrete examples within Taisei's code. For let's look at a
+piece of movement code for a fairy:
+
+.. code-block:: c
+
+   enemy->move = move_towards(VIEWPORT_W/2.0 + 200.0 * I, 0.035);
+
+The important piece here, for our purposes, is:
+
+.. code-block:: c
+
+   VIEWPORT_W/2.0 + 200.0 * I
+
+What this is doing is specifying a position on the "real" X-axis
+(``VIEWPORT_W/2.0``, which in Taisei means ``480.0 / 2.0`` or `240.0`), and
+then specifying a position on the "imaginary" Y-axis (``200.0 * I``). The ``I``
+here is the same ``i`` described in the videos and the above explanation.
+
+So what we're really looking at here is:
+
+.. code-block:: c
+
+   240.0 (real) + 200.0i (imaginary)
+
+Or "move 240 units on the (real) X-axis, and then 200 units on the (imaginary)
+Y-axis."
+
+This is what's called a **Cartesian Coordinate.** What the function
+``move_towards`` then does is make the enemy sprite/object move towards that
+point on the X/Y axis at a certain rate (defined by ``0.035``).
+
+Simple Danmaku
+^^^^^^^^^^^^^^
+
+Of course, there'd be no point in using complex numbers over a simpler X/Y
+system if it didn't provide significant advantages downstream. So let's look at
+a more complicated danmaku pattern to see why complex numbers are more
+effective for this role.
+
+.. code-block:: c
+
+   cmplx aim = cnormalize(global.plr.pos - enemy->pos);
+
+This ``aim`` variable could be passed to a ``move_towards`` function attached
+to a ``PROJECTILE`` object. The effect here would be the bullets moving
+directly towards the player in a straight line, even if the player moves around
+the screen, following the player as they move around in real-time (assuming)
+more than one bullet is fired.
+
+Let's look at the argument inside ``cnormalize`` first, ``global.plr.pos -
+e->pos``.  Both ``global.plr.pos`` and ``e->pos`` are *complex numbers*, in
+that they have both *real* and *imaginary* parts. Much like the example in
+``Simple Movement``, they represent a place on the X/Y grid.
+
+In the format of ``[X,Y]``, let's say that ``global.plr.pos`` is ``[-1, 6]``,
+and that ``enemy->pos`` is ``[6, 3]``.
+
+.. image:: images/math-01.png
+   :width: 300pt
+
+When you subtract ``[6, 3]`` (enemy position) from ``[-1, 6]`` (player
+position), you end up with ``[-7, 3]``, as seen here with ``plr->pos``.
+
+.. image:: images/math-02.png
+   :width: 300pt
+
+This also conveniently lets the enemy position ``enemy->pos`` become the new
+"origin," or ``[0, 0]``. This is useful because it means that we can more
+easily determine what angle the danmaku need to travel in to travel towards the
+player.
+
+As a vector, ``[-7, 3]``  points from the enemy position to the player position. Its
+length is the distance between enemy and player. Its direction is the direction
+we want the danmaku to travel in. In this example, we don’t care about the
```

Comment:
> Its direction is the direction we want the danmaku to travel in

It's probably better to say "angle". When we refer to a direction in code we usually mean a vector (complex) quantity, i.e. the `aim` from this example would be a "direction vector".